api/src/main/java/io/grpc/Status.java

```diff
@@ -672,4 +676,38 @@ public boolean equals(Object obj) {
   public int hashCode() {
     return super.hashCode();
   }
+
+  /**
+   * If it is a code that should not be generated by the library, convert the status into one
+   * that is allowed.  Otherwise, return the argument.
+   * @param status status as provided by server
+   * @return Transformed status or argument depending on the Status.code
+   */
+  public static Status convertServerStatus(Status status) {
```

Comment:
This seems specific to your use-case. I don't think this should be public API.

Generally we just use UNAVAILABLE. If you want to use INTERNAL some, that could be fine, but I don't think it makes sense to use UNKNOWN.

You shouldn't lose the original status, and you should also add more information to provide context. The original message will be _very_ misleading if copied directly. See https://github.com/grpc/grpc-java/pull/8953 for an example.